snuba/web/query.py

```diff
@@ -83,18 +155,24 @@ def parse_and_run_query(
     )
 
     try:
-        result = _run_query_pipeline(
-            dataset=dataset,
-            request=request,
-            timer=timer,
-            query_metadata=query_metadata,
-            robust=robust,
-            concurrent_queries_gauge=concurrent_queries_gauge,
+        result = executor(
+            dataset, request, timer, query_metadata, robust, concurrent_queries_gauge,
         )
         if not request.settings.get_dry_run():
             record_query(request, timer, query_metadata, result.extra)
     except QueryException as error:
-        record_query(request, timer, query_metadata, error.extra)
+        if isinstance(error.__cause__, QuotaExceeded):
+            meta: SnubaQueryMetadata = RateLimitedRequestMetadata(
+                request=query_metadata.request,
+                dataset=query_metadata.dataset,
+                timer=query_metadata.timer,
+                query_list=[],
+                projects=query_metadata.projects,
+            )
```

Comment:
This is the way we can force the `status` recorded to not depend on the status of the first query. As there is no query being executed.

Reply:
I noted this in the other PR, but using a separate exception class seems unnecessary since you could check to see if the quota control is enabled or not.